lib/api/routes/iterations.rb

```diff
@@ -11,17 +11,29 @@ class Iterations < Core
         require_key
 
         if current_user.guest?
```

Comment:
> Seems a little bit strange to me at a first sight, but I have no idea if this is by design or not.

Uhm, no, I can't imagine that it is. If I recall correctly, most of this code happened in a hurry as things kind of suddenly happened.

Yeah, it would be nice to consolidate things (not necessarily in this PR, I'll pull this in to fix the bug).